src/main/options.js

```diff
@@ -118,6 +120,53 @@ function inferParser(filepath, plugins) {
   const filepathParts = normalizePath(filepath).split("/");
   const filename = filepathParts[filepathParts.length - 1].toLowerCase();
 
+  let shebang = null;
+  const getShebang = () => {
+    if (shebang === null) {
+      if (typeof filepath !== "string" || filename.indexOf(".") !== -1) {
+        shebang = "";
+        return shebang;
+      }
+
+      let fd;
+      try {
+        fd = fs.openSync(filepath, "r");
+      } catch (err) {
+        shebang = "";
+        return shebang;
+      }
+
+      try {
+        const liner = new readlines(fd);
+        const firstLine = liner
+          .next()
+          .toString("utf8")
+          .trim();
+        if (!firstLine.startsWith("#!")) {
+          shebang = "";
+        } else if (firstLine.startsWith("#!/usr/bin/env")) {
+          shebang = firstLine.replace(/(#!\/usr\/bin\/env\s+(?:\S+)).*/, "$1");
```

Comment:
Does this mean that the `shebang` will never include `/usr/bin/env`?

Reply:
No, it will include `/usr/bin/env`. What I’m doing is taking

* the shebang line up to the first space, *unless* the line starts with `#!/usr/bin/env`;
* if it does start with `#!/usr/bin/env`, then take the shebang line up to the *second* space.

It may or may not be the neatest way, but anyway, the idea is to get the command but discard any arguments.

```
petter@petter:~$ node
> '#!/usr/bin/env node --foo --bar --baz'.replace(/(#!\/usr\/bin\/env\s+(?:\S+)).*/, "$1");
'#!/usr/bin/env node'
> '#!/bin/node --foo --bar'.replace(/^(\S+).*/, "$1");
'#!/bin/node'
```

Whether it *should* include `env` is another question. How do people generally write shebang lines? Would it make more sense to, say, only look at the executable name?

Reply:
Nice 👍. Thanks for explaining!